airbyte-integrations/connectors/source-appstore-singer/source_appstore_singer/source.py

```diff
@@ -0,0 +1,110 @@
+"""
+MIT License
+
+Copyright (c) 2020 Airbyte
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+import json
+
+from airbyte_protocol import AirbyteConnectionStatus
+from appstoreconnect import Api
+from base_python import AirbyteLogger
+from base_singer import SingerSource, Status
+
+from datetime import date, timedelta
+
+
+class SourceAppstoreSinger(SingerSource):
+    TAP_CMD = "tap-appstore"
+
+    def check_config(self, logger: AirbyteLogger, config_path: str, config: json) -> AirbyteConnectionStatus:
+        """
+        Tests if the input configuration can be used to successfully connect to the integration
+            e.g: if a provided Stripe API token can be used to connect to the Stripe API.
+
+        :param logger: Logging object to display debug/info/error to the logs
+            (logs will not be accessible via airbyte UI if they are not passed to this logger)
+        :param config_path: Path to the file containing the configuration json config
+        :param config: Json object containing the configuration of this source, content of this json is as specified in
+        the properties of the spec.json file
+
+        :return: AirbyteConnectionStatus indicating a Success or Failure
+        """
+        try:
+            # create request fields for testing
+            api_fields_to_test = {
+                "subscription_report": {"reportType": "SUBSCRIPTION", "frequency": "DAILY", "reportSubType": "SUMMARY", "version": "1_2"}
+            }
+            test_date = date.today() - timedelta(days=2)
```

Comment:
What will happen here if the person uses this has a new appstore account (not 2 days old)?

I'm wondering if we can make this a little easier and just check the http status code of the response. I looked at the [appstoreconnect docs](https://developer.apple.com/documentation/appstoreconnectapi/interpreting_and_handling_errors) and it looks like they use standard http codes. If we request a report for a date that hasn't been generated yet does it the API return a 200 and an empty report or does it return a 400? If it returns a 200, then we can just check that the http status code is good and not worry about trying to check that the report exists itself. Our goal in this method is to make sure that the credentials are working, not to worry about actually pulling data.

Let me know if that doesn't make sense for some reason and we can move forward without changing anything.

Reply:
Understood, let me change this.